subprojects/docs/src/docs/userguide/troubleshooting/validation_problems.adoc

```diff
@@ -398,3 +398,13 @@ Gradle uses the key to generate a name for the (sub)property.
 Only allowing keys of type `Enum`, `Integer`, and `String` guarantees that these names are unique and well-formed, and is preferable to relying on `toString()` for producing such names.
 
 To fix this problem, change the type of the key to `Enum`, `Integer`, or `String`.
+
+[[unsupported_nested_type]]
+== Unsupported nested type
+
+This error indicates that an unsupported type is annotated as nested.
+Nested types must declare annotated properties, which themselves are checked for annotations.
```

Comment:
:x: This is not actually true. Think of the case with `CommandLineArgumentProvider`... 🤔 I wonder how we could explain this correctly... Let's talk!

Reply:
I guess nested types should either have annotated properties and/or some conditional behavior where capturing the type itself as input is important.

Reply:
@lptr I agree this explanation is no longer correct. I wonder if we could state more clearly what the implementation does: 

> Nested types are expected to declare annotated properties, which themselves are checked for annotations. Accordingly, types which cannot have any annotated property (Java SE, Kotlin stdlib, GString) are not supported. All other types are expected to either declare annotated properties or some conditional behaviour where capturing the type itself as input is important.

Similarly, the reason displayed could be:

> Reason: Nested types which cannot have any annotated property are not supported.